### DISCUSSION TITLE
View JS logs after bundled and issue with useReducer

### DISCUSSION THREAD

**Discussion by mechanicalhack** [microsoft/react-native-windows#10666] [Q&A] [answered]
Hello!
I have bundled my app and I am sideloading it. I currently have a problem with useReducer where the screen that implements it is white when I navigate to it. All the other screens work just fine. The weird part is it works just fine when I run it with yarn run windows. I am currently able to see the logs from the app through Event Tracing for Windows (ETW) using the Device Portal. I am unable to find any JS logs in it though. Do I need to do something extra to get it to output the JS logs? I made some console logs just to see if I could see anything come through and I can't. I have searched for an answer on the useReducer issue as well and I haven't been able to find a solution. When I remove use reducer it works just fine as well. Any help will be much appreciated. Thanks.

**Comment by jonthysell** [comment, marked as answer]
You can specify a  native callback to run every time the JS runs `console.log`. See https://microsoft.github.io/react-native-windows/docs/next/native-api/ReactInstanceSettings#nativelogger

This defaults in Debug to the logging you're seeing, but in Release it does nothing. You can specify your own callback.

**Reply by mechanicalhack** [reply]
Thank you for the response. Do I need to specify a native callback to run every time any exception is thrown as well or does it already do that in release?

**Reply by mechanicalhack** [reply]
Also, do you by chance know of any examples of anyone else doing something like this?

**Reply by jonthysell** [reply]
I've only ever specified the native logger to to exactly what you're doing - redirect the logging while debugging issues that only appear in Release mode.

IIRC, in Debug the exceptions show the RedBox popup. You might take a look at https://microsoft.github.io/react-native-windows/docs/native-api/IRedBoxHandler for specifying your own error handling.

**Reply by mechanicalhack** [reply]
Thank you for your help

**Comment by AhmedMustafa505** [comment]
Can you @jonthysell give the full steps to add the logger in release mode, because I'm stuck in that and couldn't see the logs any where
